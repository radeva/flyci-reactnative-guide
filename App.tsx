/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React, { useState } from 'react';
import type { PropsWithChildren } from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
  TouchableOpacity,
} from 'react-native';

import {
  Colors,
  DebugInstructions,
  Header,
  LearnMoreLinks,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';

type SectionProps = PropsWithChildren<{
  title: string;
}>;

function Section({ children, title }: SectionProps): React.JSX.Element {
  const isDarkMode = useColorScheme() === 'dark';

  return (
    <View style={styles.sectionContainer}>
      <Text
        style={[
          styles.sectionTitle,
          {
            color: isDarkMode ? Colors.white : Colors.black,
          },
        ]}>
        {title}
      </Text>
      <Text
        style={[
          styles.sectionDescription,
          {
            color: isDarkMode ? Colors.light : Colors.dark,
          },
        ]}>
        {children}
      </Text>
    </View>
  );
}

function App(): React.JSX.Element {
  const isDarkMode = useColorScheme() === 'dark';
  const [counterValue, setCounterValue] = useState(0);
  const incrementValue = 2;

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  const handleIncrement = () => {
    setCounterValue(counterValue + incrementValue);
  };

  const handleDecrement = () => {
    setCounterValue(counterValue - incrementValue);
  };

  const handleReset = () => {
    setCounterValue(0);
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor={backgroundStyle.backgroundColor}
      />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        <Header />
        <View
          style={{
            backgroundColor: isDarkMode ? Colors.black : Colors.white,
          }}>
          <Section title="Step One">
            Edit <Text style={styles.highlight}>App.tsx</Text> to change this
            screen and then come back to see your edits.
          </Section>
          <Section title="See Your Changes">
            <ReloadInstructions />
          </Section>
          <Section title="Debug">
            <DebugInstructions />
          </Section>
          <Section title="Counter">
            <Text style={styles.counterValue} testID={'counterValue'}>
              {counterValue}
            </Text>
            <TouchableOpacity onPress={handleIncrement} style={styles.button}>
              <Text>Increment</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={handleDecrement} style={styles.button}>
              <Text>Decrement</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={handleReset} style={styles.button}>
              <Text>Reset</Text>
            </TouchableOpacity>
          </Section>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
  button: {
    fontSize: 16,
    padding: 13,
    borderRadius: 8,
    backgroundColor: 'dodgerblue',
    color: 'white',
  },
  counterValue: {
    fontSize: 32,
    fontWeight: 'bold',
    color: 'black',
  },
});

export default App;
