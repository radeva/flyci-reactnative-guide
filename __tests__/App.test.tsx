/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: import explicitly to use the types shipped with jest.
import { it } from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';

it('renders correctly', () => {
  renderer.create(<App />);
});

it('should render content correctly', () => {
  const appContent = renderer.create(<App />).toJSON();

  expect(appContent).toMatchSnapshot();
});

it('should increment counter', () => {
  const { getByText, getByTestId } = render(<App />);

  const counterValue = getByTestId('counterValue');
  console.log(counterValue);
  expect(counterValue.props.children).toEqual(0);
  fireEvent.press(getByText('Increment'));
  expect(counterValue.props.children).toEqual(1);
});

it('should decrement counter', () => {
  const { getByText, getByTestId } = render(<App />);

  const counterValue = getByTestId('counterValue');
  console.log(counterValue);
  expect(counterValue.props.children).toEqual(0);
  fireEvent.press(getByText('Decrement'));
  expect(counterValue.props.children).toEqual(-1);
});

it('should reset counter', () => {
  const { getByText, getByTestId } = render(<App />);

  const counterValue = getByTestId('counterValue');
  console.log(counterValue);
  expect(counterValue.props.children).toEqual(0);
  fireEvent.press(getByText('Increment'));
  expect(counterValue.props.children).toEqual(1);
  fireEvent.press(getByText('Reset'));
  expect(counterValue.props.children).toEqual(0);
});
