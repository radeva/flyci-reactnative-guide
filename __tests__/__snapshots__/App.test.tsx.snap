// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render content correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          [
            {
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            },
            {
              "backgroundColor": "#F3F3F3",
            },
          ]
        }
      >
        <Image
          accessibilityRole="image"
          source={
            {
              "testUri": "../Libraries/NewAppScreen/components/logo.png",
            }
          }
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "height": undefined,
                "width": undefined,
              },
              {
                "marginBottom": -192,
                "marginLeft": -128,
                "opacity": 0.2,
                "overflow": "visible",
                "resizeMode": "cover",
              },
            ]
          }
          testID="new-app-screen-header"
        />
        <Text
          style={
            [
              {
                "fontSize": 40,
                "fontWeight": "700",
                "textAlign": "center",
              },
              {
                "color": "#000",
              },
            ]
          }
        >
          Welcome to
          

          React Native
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFF",
          }
        }
      >
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                {
                  "color": "#000",
                },
              ]
            }
          >
            Step One
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                {
                  "color": "#444",
                },
              ]
            }
          >
            Edit 
            <Text
              style={
                {
                  "fontWeight": "700",
                }
              }
            >
              App.tsx
            </Text>
             to change this screen and then come back to see your edits.
          </Text>
        </View>
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                {
                  "color": "#000",
                },
              ]
            }
          >
            See Your Changes
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                {
                  "color": "#444",
                },
              ]
            }
          >
            <Text>
              Press 
              <Text
                style={
                  {
                    "fontWeight": "700",
                  }
                }
              >
                Cmd + R
              </Text>
               in the simulator to reload your app's code.
            </Text>
          </Text>
        </View>
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                {
                  "color": "#000",
                },
              ]
            }
          >
            Debug
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                {
                  "color": "#444",
                },
              ]
            }
          >
            <Text>
              Press 
              <Text
                style={
                  {
                    "fontWeight": "700",
                  }
                }
              >
                Cmd + D
              </Text>
               in the simulator or
               
              <Text
                style={
                  {
                    "fontWeight": "700",
                  }
                }
              >
                Shake
              </Text>
               your device to open the Dev Menu.
            </Text>
          </Text>
        </View>
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                {
                  "color": "#000",
                },
              ]
            }
          >
            Counter
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                {
                  "color": "#444",
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
              testID="counterValue"
            >
              0
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "dodgerblue",
                  "borderRadius": 8,
                  "color": "white",
                  "fontSize": 16,
                  "opacity": 1,
                  "padding": 13,
                }
              }
            >
              <Text>
                Increment
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "dodgerblue",
                  "borderRadius": 8,
                  "color": "white",
                  "fontSize": 16,
                  "opacity": 1,
                  "padding": 13,
                }
              }
            >
              <Text>
                Decrement
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "dodgerblue",
                  "borderRadius": 8,
                  "color": "white",
                  "fontSize": 16,
                  "opacity": 1,
                  "padding": 13,
                }
              }
            >
              <Text>
                Reset
              </Text>
            </View>
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
